generic/tests/ping.py

```diff
@@ -54,10 +54,11 @@ def run(test, params, env):
     ping_ext_host = params.get("ping_ext_host", "no") == "yes"
     pre_cmd = params.get("pre_cmd", None)
     vm = env.get_vm(params["main_vm"])
+    serial_status = params.get("serial_login", "no")
 
     error_context.context("Login to guest", test.log.info)
     vm.verify_alive()
-    session = vm.wait_for_login(timeout=timeout)
+    session = vm.wait_for_login(timeout=timeout, serial=serial_status)
```

Comment:
Please check the type of `serial` required.

Reply:
Updated, please help review this patch again, thanks a lot.